rdmo/projects/assets/js/interview/components/main/page/PageHeadFormModal.js

```diff
@@ -3,70 +3,96 @@ import PropTypes from 'prop-types'
 
   const ref = useRef(null)
-  const [inputValue, setInputValue] = useState('')
-  const [hasError, setHasError] = useState(false)
+
+  const initialValues = {
+    text: initial || '',
+    value: ''
+  }
+
+  const [values, setValues, initValues] = useLsState('rdmo.interview.reuse', initialValues, ['text', 'value'])
+  const [errors, setErrors] = useState([])
 
   const handleSubmit = () => {
-    if (isEmpty(inputValue) && !isNil(initial)) {
-      setHasError(true)
+    if (isEmpty(values.text)) {
+      setErrors({ text: true })
     } else {
-      onSubmit(inputValue)
+      onSubmit(values.text, values.value)
     }
   }
 
-  // update the inputValue
+  // init the form values
   useEffect(() => {
-    if (show) {
-      setInputValue(initial || '')
+    if (show && !isNil(attribute)) {
+      initValues()
     }
   }, [show])
 
   // remove the hasError flag if an inputValue is entered
   useEffect(() => {
-    if (!isEmpty(inputValue)) {
-      setHasError(false)
+    if (!isEmpty(values.text)) {
+      setErrors({ text: false })
     }
-  }, [inputValue])
+  }, [values.text])
 
   // focus when the modal is shown
   useFocusEffect(ref, show)
 
   return (
     <Modal title={title} show={show} submitLabel={submitLabel} submitProps={{className: `btn btn-${submitColor}`}}
-           onClose={onClose} onSubmit={handleSubmit} disableSubmit={hasError}>
+           onClose={onClose} onSubmit={handleSubmit} disableSubmit={errors}>
       {
-        isNil(initial) ? (
+        isNil(attribute) ? (
           <div>
             {gettext('You can add a new tab using the create button.')}
           </div>
         ) : (
-          <div className={classNames({'form-group': true, 'has-error': hasError })}>
-            <label className="control-label" htmlFor="interview-page-tabs-modal-form-title">
-              {gettext('Name')}
-            </label>
-            <input
-              ref={ref}
-              className="form-control"
-              id="interview-page-tabs-modal-form-title"
-              type="text"
-              value={inputValue}
-              onChange={(event) => setInputValue(event.target.value)}
-              onKeyPress={(event) => {
-                if (event.code === 'Enter') {
-                  handleSubmit()
-                }
-              }}
-            />
-
-            <p className="help-block">{gettext('Please give the tab a meaningful name.')}</p>
-          </div>
+          <>
+            <div className={classNames({'form-group': true, 'has-error': errors.text })}>
+              <label className="control-label" htmlFor="interview-page-tabs-modal-form-title">
+                {gettext('Name')}
+              </label>
+              <input
+                ref={ref}
+                className="form-control"
+                id="interview-page-tabs-modal-form-title"
+                type="text"
+                value={values.text}
```

Comment:
I get some kind of js warning about an `uncontrolled input `, I could resolve it by adding a optional empty string to `value=`:
```js
                value={values.text || ''}
```

Warning message:
```js
Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing
 from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input 
element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components   
...

PageHeadFormModal@webpack://rdmo/./rdmo/projects/assets/js/interview/components/main/page/  PageHeadFormModal.js?:38:15

```

Reply:
This is a bit weird since values.text is set from `initialValues`. Can you reproduce this? It might hint to something with the local storage.

Reply:
When I refresh and am on a page in the interview, I open the the new dataset tab and this warning pops up.
When I do the same with this "fix", the warning does not pop up.

Reply:
I think I found a fix for the `useLsState` hook. It is now a bit more readable I hope. I basically does the same when first calling the hook and when calling `initValues` (the third return value). If it is an object it merges the `initialValues` with the state from the LS (omitting `omit`) if it is a single value (e.g. a string) it will return the LS value if it set or the initial value. It think this is what one expect. The latter case is not used in RDMO, yet.

Reply:
thanks, it's working now and I don't get a warning.